pkg/cidata/cidata.go

```diff
@@ -370,6 +372,54 @@ func GenerateISO9660(instDir, name string, y *limayaml.LimaYAML, udpDNSLocalPort
 	return iso9660util.Write(filepath.Join(instDir, filenames.CIDataISO), "cidata", layout)
 }
 
+// getHostTimezone tries to get host timezone from env or specific files
+func getHostTimezone() string {
+	if runtime.GOOS == "windows" {
+		logrus.Debug("skip get host timezone on Windows")
+		return ""
+	}
+
+	// 1. try to get timezone from env.TZ
+	tzEnv := os.Getenv("TZ")
+	if tzEnv != "" &&
+		// skip if TZ is in "std offset dst offset, rule" format
+		!strings.Contains(tzEnv, " ") {
+		return tzEnv
+	}
+	logrus.Debugf("cannot get timezone from env TZ: [%s]", tzEnv)
+
+	// 2. try to get timezone from /etc/timezone
+	tzName, err := os.ReadFile("/etc/timezone")
+	if err == nil {
+		return string(tzName)
+	}
+	logrus.Debugf("cannot get timezone from /etc/timezone: %s", err)
+
+	// 3. try to get timezone from /etc/localtime soft link
+	// base on the assumption that <zoneinfo_basedir>/zone.tab file exists
+	// and /etc/localtime is a soft link to <zoneinfo_basedir>/<timezone>
+	tzFilePath, err := filepath.EvalSymlinks("/etc/localtime")
+	if err == nil {
+		for baseDir := filepath.Dir(tzFilePath); baseDir != "/"; baseDir = filepath.Dir(baseDir) {
+			anchorFile := filepath.Join(baseDir, "zone.tab")
+			_, err = os.Stat(anchorFile)
+			if err != nil {
+				continue
+			}
+
+			tz := strings.TrimPrefix(tzFilePath, baseDir+"/")
+			_, err = time.LoadLocation(tz)
+			if err == nil {
+				return tz
+			}
```

Comment:
I don't think the check if `time.LoadLocation` succeeds is useful. It is the setting that has been configured, so if the datafile somehow got corrupted, it is not our problem. And the real issue would be if the data file exists inside the VM, and we can't answer that anyways.

If we kept the check, then I would still argue that we need to `return ""` in the failure case, and not to keep looking in higher levels of the directory tree; that seems to be wrong either way.

```suggestion
			return strings.TrimPrefix(tzFilePath, baseDir+"/")
```